tsdb/engine/tsm1/batch_integer_test.go

```diff
@@ -910,9 +910,8 @@ func TestIntegerArrayDecodeAll_MinMax(t *testing.T) {
 
 func TestIntegerArrayDecodeAll_Quick(t *testing.T) {
 	quick.Check(func(values []int64) bool {
-		exp := values
```

Comment:
Felt like I was following along pretty well until I hit these test changes in the tsm package 😃 . What's going on with these?

Reply:
These are all changes to appease `staticcheck` (the version used by the CLI is newer, so it got upgraded when I added the CLI dependency and uncovered a new set of issues).

The old code would:
1. Set `exp = values`
2. Set `exp = empty` if values is `nil`
3. Iterate over `values`

staticcheck complained about step 3 because step 2 shows `values` could potentially be `nil` and cause a panic in the `range`. An alternate fix would have been to update 3 to iterate over `exp` instead of `values`.